src/dcast.d

```diff
@@ -97,8 +97,17 @@ extern (C++) Expression implicitCastTo(Expression e, Scope* sc, Type t)
                     //printf("type %p ty %d deco %p\n", type, type.ty, type.deco);
                     //type = type.semantic(loc, sc);
                     //printf("type %s t %s\n", type.deco, t.deco);
-                    e.error("cannot implicitly convert expression (%s) of type %s to %s",
-                        e.toChars(), e.type.toChars(), t.toChars());
+                    immutable(char)* arrayHint = "";
+                    if (t.ty == Tarray)
+                    {
+                        auto ad = isAggregate(e.type);
+                        if (ad && ad.isInputRange())
+                        {
+                            arrayHint = ", use .array (in std.array) to force allocation of range elements";
+                        }
```

Comment:
I just don't think this is a good idea for such messages, it is too hard to guess what the user might be trying to accomplish. Guess wrong, and things can be very confusing.